File: ts24-projects/ts-account/src/main/java/vn/conyeu/ts/domain/Question.java
package vn.conyeu.ts.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;
import vn.conyeu.common.domain.LongId;

//@formatter:off
@Entity @Table
@Getter @Setter @NoArgsConstructor
@DynamicInsert @DynamicUpdate
@AttributeOverride(name = "id", column = @Column(name = "questionId"))
//@formatter:on
public class Question extends LongId<Question> {

    @Column(length = 300, nullable = false)
    private String title;

    @Column(length = 2000, nullable = false)
    private String reply;

    @Column(length = 50, nullable = false)
    private String softType;

    @ColumnDefault("0")
    private Boolean shared;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", nullable = false)
    private TsUser user;


}

File: ng-project/TsApp/projects/kpi-app/src/app/models/question.ts
import { BaseModel } from "./base-model";
import { JsonObject } from "./common";

export class Question extends BaseModel {
    question_id?: number;
    reply?: string;
    title?: string;
    soft_type?: string;
    shared?: boolean;
    user_id?: number;

  static from(data: JsonObject): Question {
    return new Question().update(data);
  }
}
